lib/std/Build.zig

```diff
@@ -1285,10 +1285,14 @@ pub fn standardTargetOptions(b: *Build, args: StandardTargetOptionsArgs) Resolve
 }
 
 pub fn parseTargetQuery(options: std.Target.Query.ParseOptions) error{ParseFailed}!std.Target.Query {
-    var diags: Target.Query.ParseOptions.Diagnostics = .{};
-    var opts_copy = options;
-    opts_copy.diagnostics = &diags;
-    return std.Target.Query.parse(options) catch |err| switch (err) {
+    var diags_dummy: std.Target.Query.ParseOptions.Diagnostics = undefined;
+    var diags: *std.Target.Query.ParseOptions.Diagnostics = options.diagnostics orelse &diags_dummy;
+    var opts_with_diags = options;
```

Comment:
These two don't need to be `var` and the compiler might be able to infer the type of `diags`.

Reply:
`opts_with_diags ` cannot be const as setting the diagnostics field may or may not happen.

Will convert diags to const and inferred type.